src/polygonarea.rs

```diff
@@ -1 +1,646 @@
+            return 1;
+        } else if lon12 < 0.0 && lon1 >= 0.0 && lon2 < 0.0 {
+            return -1;
+        } else {
+            return 0;
+        }
+    }
+
+    fn transitdirect(lon1: f64, lon2: f64) -> i64 {
+        // We want to compute exactly: floor(lon2 / 360) - floor(lon1 / 360)
+        let lon1 = lon1 % 720.0;
+        let lon2 = lon2 % 720.0;
+
+        let a = if 0.0 <= lon2 && lon2 < 360.0 {
+            0
+        } else {
+            1
+        };
+
+        let b = if 0.0 <= lon1 && lon1 < 360.0 {    
+            0
+        } else {
+            1
+        };
+
+        return a - b;
+    }
+
+    fn reduce_area(&self, area: f64, signed: bool) -> f64 {
+        let geoid_area = self.geoid.area(); // Area of the planet
+        let mut area = area % geoid_area;
+
+        // Translation of the following cpp code:
+        // if (crossings & 1) area += (area < 0 ? 1 : -1) * _area0/2;
+        if self.crossings % 2 != 0 {
+            if area < 0.0 {
+                area += geoid_area / 2.0;
+            } else {
+                area -= geoid_area / 2.0;
+            }
+        }
+
+        // Area is with the clockwise sense. If needed convert to counter-clockwise convention.
+        area = match self.winding {
+            Winding::Clockwise => area,
+            Winding::CounterClockwise => -area,
+        };
+
+        if signed {
+            // Put area in (-geoid_area/2, geoid_area/2]
+            if area > geoid_area / 2.0 {
+                area -= geoid_area;
+            } else if area <= -geoid_area / 2.0 {
+                area += geoid_area;
+            }
+        } else {
+            // Negative values mean the polygon is larger than half the planet. Correct for this.
+            if area < 0.0 {
+                area += geoid_area;
+            }
+        }
+
+        area
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::Geodesic;
+    use approx::assert_relative_eq;
+
+    #[test]
+    fn test_simple_polygonarea() {
+        let geoid = Geodesic::wgs84();
+        let mut pa = PolygonArea::new(&geoid, Winding::CounterClockwise);
+
+        pa.add_point(0.0, 0.0);
+        pa.add_point(0.0, 1.0);
+        pa.add_point(1.0, 1.0);
+        pa.add_point(1.0, 0.0);
+
+        let (perimeter, area, _count) = pa.compute(true);
+
+        assert_relative_eq!(perimeter, 443770.917, epsilon = 1.0e-3);
+        assert_relative_eq!(area, 12308778361.469, epsilon = 1.0e-3);
+
+        let mut pa = PolygonArea::new(&geoid, Winding::Clockwise);
+
+        pa.add_point(0.0, 0.0);
+        pa.add_point(0.0, 1.0);
+        pa.add_point(1.0, 1.0);
+        pa.add_point(1.0, 0.0);
+
+        let (perimeter, area, _count) = pa.compute(true);
+
+        assert_relative_eq!(perimeter, 443770.917, epsilon = 1.0e-3);
+        assert_relative_eq!(area, -12308778361.469, epsilon = 1.0e-3);
+    }
+
+    #[test]
+    fn test_edge_2() {
```

Comment:
What are you testing here?

Reply:
Oops!  That was a halfway written test that I realized that I didn't need and forgot to delete it.   I've now deleted it.